### PR TITLE
python310Packages.flake8-future-import: fix build

### PR DESCRIPTION
**Pull request by wamserma**
###### Motivation for this change

https://github.com/xZise/flake8-future-import/issues/24
ZHF: #144627
ping @NixOS/nixos-release-managers

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 144650"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Please target staging

**Comment by risicle** [comment]
`python310Packages.flake8-future-import` building successfully here, macos 10.15.

**Comment by risicle** [comment]
@ofborg eval

**Comment by jonringer** [comment]
> re-targeted to staging, should be backported after branch-off

If it's fixing regressions which already exist, they should be merged.

The "breaking changes to staging" block is to prevent new regressions from entering

**Comment by wamserma** [comment]
> > re-targeted to staging, should be backported after branch-off
> 
> If it's fixing regressions which already exist, they should be merged.
> 
> The "breaking changes to staging" block is to prevent new regressions from entering

It's an un-breaking PR, but as it includes a bump of `python3.flake8` we have quite a number of rebuilds.

Let the @NixOS/nixos-release-managers decide where they prefer this PR. Feel free to re-target as required before merging.

**Comment by SuperSandro2000** [comment]
> we have quite a number of rebuilds.

That does not matter for the breaking change policy. It is about not bumping compilers like gcc to a breaking version.

> Let the @NixOS/nixos-release-managers decide where they prefer this PR. Feel free to re-target as required before merging.

jonringer is the release manager.

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.flake8-future-import python310Packages.flake8 python39Packages.flake8-future-import python39Packages.flake8

**Review by jonringer — CHANGES_REQUESTED** [review]
This is causing breakages with the manual. the flake8 bump will need to wait til after ZHF

https://github.com/NixOS/nixpkgs/runs/4137245989?check_suite_focus=true

**Comment by wamserma** [comment]
@jonringer (hopefully) fixed

File: pkgs/development/python-modules/pytest-flake8/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pytest-flake8/default.nix:24]: ´Please use fetchPatch and inline it into patches.

**Comment by SuperSandro2000** [comment]
@ofborg eval
@ofborg build

**Comment by SuperSandro2000** [comment]
I think it is failing due to coreutils.

**Comment by wamserma** [comment]
> I think it is failing due to coreutils.

Found this in the logs if the coreutils checks:

```
SKIP: tests/du/2g
=================

2g.sh: skipped test: very expensive: disabled by default
This test is very expensive, so it is disabled by default.
To run it anyway, rerun make check with the RUN_VERY_EXPENSIVE_TESTS
environment variable set to yes.  E.g.,

  env RUN_VERY_EXPENSIVE_TESTS=yes make check

or use the shortcut target of the toplevel Makefile,

  make check-very-expensive

./tests/init.sh: line 106: printf: write error: Broken pipe
SKIP tests/du/2g.sh (exit status: 77)

FAIL: tests/du/basic
====================

--- exp 2021-11-09 11:58:41.313614539 +0000
+++ out 2021-11-09 11:58:41.278614356 +0000
@@ -1,7 +1,7 @@
-9      d/sub/2
-10     d/sub
-9      d/1
-20     d
+1      d/sub/2
+2      d/sub
+1      d/1
+4      d
 ===
-10     d/sub
-10     d
+2      d/sub
+2      d
FAIL tests/du/basic.sh (exit status: 1)
```

So it might be an intermittent error.

**Comment by wamserma** [comment]
Rebased, now
@ofborg eval
@ofborg build

**Comment by wamserma** [comment]
@SuperSandro2000 fixed

**Comment by SuperSandro2000** [comment]
@jonringer I think we can merge this now. What do you think?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 144650` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>home-assistant: log was empty</li>
  </ul>
</details>

**Comment by trofi** [comment]
Does patch fetch work for you? Fails for me as:

```
$ nix build -f. python3Packages.pytest-flake8 -L

eda4ef74c0f25b856fe282742ea206b21e94c24c.patch> trying https://github.com/tholo/pytest-flake8/commit/eda4ef74c0f25b856fe282742ea206b21e94c24c.patch
eda4ef74c0f25b856fe282742ea206b21e94c24c.patch>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
eda4ef74c0f25b856fe282742ea206b21e94c24c.patch>                                  Dload  Upload   Total   Spent    Left  Speed
eda4ef74c0f25b856fe282742ea206b21e94c24c.patch> 100  3155  100  3155    0     0  37639      0 --:--:-- --:--:-- --:--:-- 37559
error: hash mismatch in fixed-output derivation '/nix/store/qnplqiw1cs5c73099virwa7i60krzq9z-eda4ef74c0f25b856fe282742ea206b21e94c24c.patch.drv':
         specified: sha256-/QBKi1nR5pCCEvODf4cmEyxdXihLLJgRUljDloKPLAA=
            got:    sha256-kfll+I9akZM4Vp62Q1M9r2Mvs2TM2Lm5mRkA3aDmAE8=
error: 1 dependencies of derivation '/nix/store/1vjdyv87g6fjdh93p9dascwy8p984xyy-python3.9-pytest-flake8-1.0.7.drv' failed to build
```

**Comment by wamserma** [comment]
@trofi When I build from current master it simply pulls `/nix/store/2743gl0snph4z967yi1kqvaaiqp75cg6-python3.9-pytest-flake8-1.0.7` from the binary cache.

```
> nix-prefetch-url https://github.com/tholo/pytest-flake8/commit/eda4ef74c0f25b856fe282742ea206b21e94c24c.patch
[0.0 MiB DL]
path is '/nix/store/2h9wk9cicphjzpbpg0pp3za25mq5l4dw-eda4ef74c0f25b856fe282742ea206b21e94c24c.patch'
001ciy19dhsqa88rhb2b51g5sb0k4s3pz0zk2a191rnib65ll07x
```

**Comment by trofi** [comment]
> nix-prefetch-url https://github.com/tholo/pytest-flake8/commit/eda4ef74c0f25b856fe282742ea206b21e94c24c.patch

Ah, that explains it. `fetchpatch` (as opposed to fetchurl` does mangle patch a bit before calculating the checksum.

Proposed hash fix as https://github.com/NixOS/nixpkgs/pull/145786

**Comment by wamserma** [comment]
Right, just didn't show up, because the patch was already fetched.

**Pull request merged**
